src/main/java/com/google/sps/data/UrlRequest.java

```diff
@@ -0,0 +1,65 @@
+package com.google.sps.data;
+
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.net.URLEncoder;
+import java.net.URL;
+import java.net.HttpURLConnection;
+import java.util.Map;
+
+/* Class conatining helper methods to make url requests */
+public final class UrlRequest {
+  /**
+   * Returns the a json string with the API response given an URL path and the query parameters.
+   */
+  public static String urlQuery(String basePath, Map<String, String> parameters) throws IOException {
+    String path = String.format("%s?%s", basePath, getParamsString(parameters));
+    URL url = new URL(path);
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+    connection.setRequestMethod("GET");
+    connection.setRequestProperty("Accept", "application/json");
+    connection.setRequestProperty("Content-Type", "application/json");
+    int responseCode = connection.getResponseCode();
+    InputStreamReader inputStream = new InputStreamReader(connection.getInputStream());
+    StringBuilder contentBuilder = new StringBuilder();
+    try (BufferedReader inputReader = new BufferedReader(inputStream)) {
```

Comment:
If you are going to wrap this in 'try' block you should have a corresponding catch.

Also, you probably want to put connection.disconnect() in a 'finally' block to ensure you disconnect even if an exception is thrown.

Reply:
Do I need the catch block even if it is a try-with-resources statement?

Reply:
It depends on how you want to handle the case of the BufferedReader failing to open.  If it fails, there will be an IOException thrown.  You probably want to catch that at some point and handle it.  

I would recommend handling the Exception as soon as possible, although I'm not sure what the best design would be.  The best design depends on how this shows up in the UI and I would have to see an example of the website in action to decide how I would handle this.  For now, you might want to just create some kind of alert or pop-up to inform the user.

Reply:
For now, I'll return an empty string.